### ISSUE TITLE
OneLogin SAML documentation is confusing

### ISSUE THREAD

**Issue by slimsag** [sourcegraph/sourcegraph#6048]
It took me an hour to setup OneLogin SAML with Sourcegraph starting from https://docs.sourcegraph.com/admin/auth

Major pain points for me include:

- The documentation tries to cover a ton of different authentication modes and providers, it's not clear to me which parts I need to pay attention to or not.
- Starting at the top of the page, I don't know if Sourcegraph supports OneLogin authentication or not. Unless someone told me it was a SAML provider, I wouldn't know. I'm not an authentication expert.
- Once I do end up on the SAML section, it's not clear what steps I actually need to perform. It says just:

> OneLogin - Administration > Apps > Add Apps > SAML Test Connector (SP)

This is not enough to make performing this task easy. We even call this out ourselves:

> Register Sourcegraph as a SAML Service Provider in the external SAML Identity Provider (such as one of those listed above). Use the following settings (___the exact names and labels vary across services___).

- "SAML Test Connector (SP)" does not exist as an option in OneLogin anymore.
- There are multiple other options in OneLogin that are not mentioned by Sourcegraph docs and I don't know if I need to configure them or not.
- If you land on our OneLogin SAML docs via Google, the page is extremely poor and doesn't explain anything: https://docs.sourcegraph.com/admin/auth/saml_with_onelogin

In short, there needs to be a guide to configuring Sourcegraph to use OneLogin if we want this to be easy to do and not take an hour to perform. It needs to be specific to OneLogin, not some broad generic document covering all authentication providers. It needs to include screenshots about what I should be trying to do.

**Comment by tsenart** [comment]
@beyang: I believe you did a lot of work on this documentation recently. Do you think @slimsag's concerns are addressed already after that?
